src/services/database.py

```diff
@@ -14,20 +15,194 @@ def get_db_connection(db_file: str):
 
     Keyword arguments:
-    erase -- flag indicating whether the database should be erased
+    reste -- flag indicating whether the database should be erased
     '''
     with get_db_connection(USERS_DB) as con:
-        if erase:
+        if reset:
             con.execute('DROP TABLE IF EXISTS users')
         con.execute('''CREATE TABLE IF NOT EXISTS users
                         (id INTEGER PRIMARY KEY AUTOINCREMENT,
                             username TEXT UNIQUE,
                             hashed_password TEXT)''')
         con.commit()
+
+def user_exists(username: str) -> bool:
+    """Returns true if user with input username exists in users database.
+
+    Args:
+        username (str): username to be checked
+
+    Returns:
+        bool: true if user exists in users database with input username otherwise false
+    """
+    with get_db_connection(USERS_DB) as con:
+        if con.execute('SELECT username FROM users WHERE username = ?', (username,)).fetchone():
+            return True
+        else:
+            return False
+
+def add_new_user(username: str, hashed_password: str) -> None:
+    """Inserts record for new user in users database.
+
+    Args:
+        username (str): username of new user
+        hashed_password (str): hashed password of new user
+    """
+    with get_db_connection(USERS_DB) as con:
+        con.execute('INSERT INTO users (username, hashed_password) VALUES (?, ?)',
+                    (username, hashed_password))
+        con.commit()
+
+def get_hashed_password(username: str) -> str:
+    """Returns hashed password of user associated with username. Returns None if user does not exists.
+
+    Args:
+        username (str): _description_
+
+    Returns:
+        str: _description_
+    """
+    if user_exists(username):
+        with get_db_connection(USERS_DB) as con:
+            user = (
+                con
+                .execute('SELECT hashed_password FROM users WHERE username = ?', (username,))
+                .fetchone()
+            )
+            return user['hashed_password']
+    else:
+        return None
+
+def initialize_courses_db(update: bool = False) -> None:
+    """Loads in course records from COURSES_SQL if update is set to true or if table does not exist.
+
+    Args:
+        update (bool, optional): If true, courses database is reloaded. Defaults to False.
+    """
+    table_exists = False
+    with get_db_connection(COURSES_DB) as con:
+        courses = con.execute('''
+                              SELECT name FROM sqlite_master WHERE type='table' AND name='courses'
+                              ''')
+        if courses.fetchone():
+            table_exists = True
+    if update or not table_exists:
+        sql_script = ''
+        with open(COURSES_SQL, 'r', encoding='utf-8') as file:
+            sql_script = file.read()
+        with get_db_connection(COURSES_DB) as con:
+            con.executescript(sql_script)
```

Comment:
initialize_from_sql can be a common place for all db creation step. Again, not everyone should be calling it.